cluster-diagnosis/utils.py

```diff
@@ -131,6 +133,100 @@ def get_pod_config(pod_name):
     return output
 
 
+PodStatus_ = collections.namedtuple('PodStatus',
+                                    'name ready_status status node_name')
+
+
+class PodStatus(PodStatus_):
+    """ A namedtupe with the following elements in this order.
+        name (string): name of the pod.
+        ready_status (string): the ready status of the pod.
+        status (string): the status of the pod (e.g. Running).
+        node_name (string): the name of the node.
+    """
+    pass
+
+
+def get_pods_summarized_status_iterator(pod_name_substring, must_exist=True):
+    """Returns a summarized status of the pods by retrieving the status
+    multiple times.
+
+    This helps avoid any false negatives that can occur
+    in the scenario wherein the status is checked just after a pod restart.
+
+    Args:
+        pod_name_substring - the substring containing the pod name.
+        must_exist - boolean to indicate that a pod with that name must exist.
+            If the condition isn't satisfied, an error will be logged.
+
+    Returns:
+        An object of type PodStatus.
+    """
+    pod_status_map = {}
+    for attempt in range(0, 10):
+        # These retry attempts will take some time. Provide some form of
+        # visual feedback to the user.
+        # Cannot use log as it'll print on a new line every time.
+        sys.stdout.write('.')
+        sys.stdout.flush()
+        for pod_status in \
+                get_pods_status_iterator(pod_name_substring, must_exist):
+            status_verdict = 'Running'
+            try:
+                temp_pod_status = get_pod_status(pod_status.name)
+                if (temp_pod_status.status != "Running"):
```

Comment:
It would be good to define constants for those status strings.

Reply:
Done.